csharp/src/Microsoft.ML.OnnxRuntime/InferenceSession.shared.cs

```diff
@@ -1044,6 +1046,130 @@ public ulong ProfilingStartTimeNs
             }
         }
 
+        private static void OrtCallback(IntPtr userData, IntPtr[] ouputs, uint numOutputs, IntPtr status)
+        {
+            var hostHdl = GCHandle.FromIntPtr(userData);
+            CallbackHost host = (CallbackHost)hostHdl.Target;
+            try
```

Comment:
if hostHdl is an `IDisposable`, use `using` instead of `try`-`finally`

Reply:
It is not disposable by design.